### ISSUE TITLE
research possibility to restrict use of functions on templates

### ISSUE THREAD

**Issue by drfaustusfade** [drfaustusfade/ngm-reportDesk#61]
using functions in expressions could cause performance issues, as the functions are run on change detection
we have been using it extensively as it is easy to do, by the cost of minor performance losses

to check if there is a suitable solution to refactor

**Comment by fakhrihawari** [comment]
This is include function that passes to. widget?

**Comment by drfaustusfade** [comment]
yes, all functions, that rerun on digest cycle, I assume that won't be possible to fix many of them but to research and measure the effect would be interesting

**Comment by fakhrihawari** [comment]
experiment branch:
https://github.com/fakhrihawari/ngm-reportHub/tree/trying-remove

**Comment by drfaustusfade** [comment]
yes, exactly, the case

> experiment branch:
> https://github.com/fakhrihawari/ngm-reportHub/tree/trying-remove
